fdbserver/ConfigBroadcaster.actor.cpp

```diff
@@ -151,6 +172,9 @@ class ConfigBroadcasterImpl {
 	Future<Void> setSnapshot(Snapshot&& snapshot, Version snapshotVersion) {
 		this->snapshot = std::forward<Snapshot>(snapshot);
 		this->lastCompactedVersion = snapshotVersion;
+		for (const auto& client : clients) {
+			actors.add(brokenPromiseToNever(pushSnapshot(this, snapshotVersion, client)));
```

Comment:
@sfc-gh-tclinkenbeard I think there might be a potential ordering issue here. [applySnapshotAndChanges](https://github.com/apple/foundationdb/blob/bd0a04d4bc462d0ab3ec8e6acb375724db3ec829/fdbserver/ConfigBroadcaster.actor.cpp#L210) calls `setSnapshot` followed immediately by `addChanges`. Since `addChanges` doesn't wait for `setSnapshot` to complete, I think incremental change messages could reach workers before snapshot messages. One solution is to have `addChanges` depend on the futures return `setSnapshot` -- do you know a good way to go about this?

Reply:
Yeah, I think `setSnapshot` should return a future that waits on all `pushSnapshot` actors. `applySnapshotAndChanges` can spawn an actor that sequentially runs `setSnapshot`, then `addChanges`. This won't require making any changes to the `addChanges` implementation.